### PR TITLE
Feature/jenkins deploy

### PR DESCRIPTION
**Pull request by JoeMcGrath**
initial stab at terraforming a management zone
It is probably a bit clunky - given first time using terraform!
comments would be pretty useful at this stage

### CODE REVIEW

File: shared/mgmt-zone/mgmtzonetemplate.tf [hunk 1]
```diff
@@ -0,0 +1,445 @@
+# The purpose of this terraform runbook is to stand up the following resources:
```
Comment (glenjamin) [inline shared/mgmt-zone/mgmtzonetemplate.tf]: I reckon this should go in an `examples` folder.

File: shared/mgmt-zone/mgmtzonetemplate.tf [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+#
+
+
+terraform {
+    required_version = ">= 0.9.2"
+    backend "azure" {
+        resource_group_name = "webops"
+        storage_account_name = "nomsstudiowebops"
+        container_name = "terraform"
+        key = "mgmt-test.terraform.tfstate"
```
Comment (glenjamin) [inline shared/mgmt-zone/mgmtzonetemplate.tf]: i'd keep this commented out in the template, as anyone using should be giving it their own name

File: shared/mgmt-zone/mgmtzonetemplate.tf [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+  }
+
+  storage_os_disk {
+    name              = "${var.jump-server-name}"
+    caching           = "ReadWrite"
+    create_option     = "FromImage"
+    managed_disk_type = "Standard_LRS"
+  }
+
+  storage_data_disk {
```
Comment (glenjamin) [inline shared/mgmt-zone/mgmtzonetemplate.tf]: does jump need a data disk?

File: shared/mgmt-zone/mgmtzonetemplate.tf [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+        virtualNetworkName: "${azurerm_virtual_network.mgmt-vnet.name}"
+        subnetName = "${azurerm_subnet.appGatewaySubnet.name}"
+        vnetID = "${azurerm_virtual_network.mgmt-vnet.id}"
+        keyVaultId = "${azurerm_key_vault.vault.id}"
+        keyVaultCertName = "${var.keyvault-cert-name}"
+        service = "${var.tags["Service"]}"
+        environment = "${var.tags["Environment"]}"
+        role = "other"
+    }
+    output "appgwcname" {
```
Comment (glenjamin) [inline shared/mgmt-zone/mgmtzonetemplate.tf]: looks like a syntax error here, output should be top-level and the resource isn't closed

File: shared/mgmt-zone/var-template.tf [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    default = "changeme"
+}
+
+// These AD ObjectIDs were found via `az ad sp list`
+variable "azure_tfuser_oid" {
+    type = "string"
+    default = "changeme"
+}
+ 
+provider "azurerm" {
```
Comment (glenjamin) [inline shared/mgmt-zone/var-template.tf]: can use the symlinks or just omit this from the example

File: shared/mgmt-zone/mgmtzonetemplate.tf (hunk 1 above)
Comment (JoeMcGrath) [inline shared/mgmt-zone/mgmtzonetemplate.tf]: done

File: shared/mgmt-zone/mgmtzonetemplate.tf (hunk 3 above)
Comment (JoeMcGrath) [inline shared/mgmt-zone/mgmtzonetemplate.tf]: oops - copy paste :) will remove

File: shared/mgmt-zone/mgmtzonetemplate.tf (hunk 4 above)
Comment (JoeMcGrath) [inline shared/mgmt-zone/mgmtzonetemplate.tf]: fixed

**Pull request merged**